pkg/jobs/execution_detail_utils.go

```diff
@@ -20,15 +20,28 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobsprofiler/profilerconstants"
 	"github.com/cockroachdb/cockroach/pkg/sql/isql"
-	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
-	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
 	"github.com/cockroachdb/errors"
 	"github.com/klauspost/compress/gzip"
 )
 
 const bundleChunkSize = 1 << 20 // 1 MiB
 const finalChunkSuffix = "#_final"
 
+var filenameToProtobinStringer map[string]func(content []byte) (string, error)
+
+// RegisterProtobinFileStringer registers a method that converts the byte
+// contents of filename to its string representation.
+func RegisterProtobinFileStringer(filename string, stringer func(content []byte) (string, error)) {
```

Comment:
Why do we need to register versus just implementing an interface?

Reply:
hmm who would implement the interface? Were you thinking we'd unmarshal the `binpb` bytes into a proto type and then call `String()` on that? I wanted to avoid this util file having to know about every `binpb` proto we end up supporting which is why I went with the register model.

Reply:
I was imagining the proto itself would, and yeah, we'd unmarshal and then call the function. Could just put the pb's already registered name in the filename? have the convention be anything that ends in `.pbbin` has, immediately prior to that, the proto's registered name (that pb_to_json would use) e.g.
```func WriteProtoBinFile(name string, msg protoutil.Messaage ) { name := fmt.Sprintf("%s_%s.pbbin", name, proto.MessageName(msg)), ...}```

and then for stringing during reads, we just 
``` protoreflect.DecodeMessage(strings.TrimSuffix(fileName[strings.LastIndex(fileName, "_")+1:], ".pbbin")```
then assert the message implements our interface and call it?

Reply:
👍 this sounds good to me, ill fix it up

Reply:
nb: I ninja edited this to use ~ instead of _ as the separator since _ can appear in message names (think embedded messages) but ~ can't.

Reply:
👍 done, rfal